src/main/java/com/google/sps/data/Note.java

```diff
@@ -1,16 +1,135 @@
 package com.google.sps.data;
 
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageSource;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.Symbol;
+import com.google.cloud.vision.v1.Word;
+import com.google.cloud.vision.v1.Paragraph;
+import com.google.cloud.vision.v1.Block;
+import com.google.cloud.vision.v1.Page;
+import com.google.cloud.vision.v1.TextAnnotation;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+
 public final class Note {
+    private long id;
+    private final String imageUrl;
+    private final String message;
+
+    // how I'd imagine the NLP text can be sorted
+    // key is the keyword/heading
+    // value is a list of related sentences
+    private HashMap<String, List<String>> categorizedText;
+
+    // Not sure if we still need this, but leaving for now
+    public Note(long id, String imageUrl, String message) {
+        this.id = id;
+        this.imageUrl = imageUrl;
+        this.message = message;
+    }
+
+    public Note(String imageUrl) throws IOException{
+        this.imageUrl = imageUrl;
+        this.message = detectDocumentText(this.imageUrl);
+    }
+
+    public String getOriginalImageUrl(){
+        return imageUrl;
+    }
+    public String getMessage(){
+        return message;
+    }
```

Comment:
Should we also create an editMessage method where the user can modify the message if it is wrong? Or for our MVP we could just assume only one person uses this webapp

Reply:
I think that's what @oulily  is working on this week?

Reply:
Oops the last question should've been in the other comment. I don't know how it got here, I'll edit the comments to fix it

Reply:
Yeah once this code is merged I can add that method to the class